packages/sanity/src/core/releases/tool/detail/documentTable/extractSearchableText.ts

```diff
@@ -0,0 +1,52 @@
+import {toPlainText} from '@portabletext/react'
+import {isPortableTextBlock} from '@portabletext/toolkit'
+
+/**
+ * Extracts a searchable string from a field value, handling various data types
+ * @param value - The field value to extract text from
+ * @returns A string representation suitable for search
+ */
+export function extractSearchableText(value: unknown): string {
+  if (typeof value === 'undefined' || value === null) return ''
+
+  // Handle primitive types
+  if (typeof value === 'string') return value
+  if (typeof value === 'number') return value.toString()
+
+  // Handle arrays
+  if (Array.isArray(value)) {
+    // Check if it's portable text
+    if (value.length > 0 && isPortableTextBlock(value[0])) {
```

Comment:
think this could give false negatives – not uncommon for portable text arrays to have a non-block as the first element. Wonder if its better to check if value is an object and has _type=="block", and extract text instead: `if (value._type === 'block') return block.children.map(child => child.text).join('')`